gossip/src/cluster_info.rs

```diff
@@ -2272,26 +2272,25 @@ impl ClusterInfo {
         thread_pool: &ThreadPool,
     ) -> Result<(), GossipError> {
         const RECV_TIMEOUT: Duration = Duration::from_secs(1);
-        fn count_packets_received(packets: &PacketBatch, counts: &mut [u64; 7]) {
+        fn count_dropped_packets(packets: &PacketBatch, dropped_packets_counts: &mut [u64; 7]) {
             for packet in packets {
                 let k = packet
                     .data(..4)
                     .and_then(|data| <[u8; 4]>::try_from(data).ok())
                     .map(u32::from_le_bytes)
                     .filter(|&k| k < 6)
                     .unwrap_or(/*invalid:*/ 6) as usize;
-                counts[k] += 1;
+                dropped_packets_counts[k] += 1;
             }
         }
-        let mut counts = [0u64; 7];
+        let mut dropped_packets_counts = [0u64; 7];
```

Comment:
It would be nice to make a struct for these. This is bound to explode at some point.

Reply:
The struct is indeed `self.stats`; see `record_dropped_packets`.
But we need to map the enum identifier (u32) to the struct fields somewhere and that is what the combination of `dopped_packet_counts` and `record_dropped_packets` try to do.